runconfig/hostconfig_unix.go

```diff
@@ -51,10 +52,17 @@ func validateQoS(hc *container.HostConfig) error {
 // cpu-rt-runtime and cpu-rt-period can not be greater than their parent, cpu-rt-runtime requires sys_nice
 func validateResources(hc *container.HostConfig, si *sysinfo.SysInfo) error {
 	if (hc.Resources.CPURealtimePeriod != 0 || hc.Resources.CPURealtimeRuntime != 0) && !si.CPURealtime {
-		return fmt.Errorf("Your kernel does not support CPU real-time scheduler")
+		return errdefs.InvalidParameter(fmt.Errorf("Your kernel does not support CPU real-time scheduler"))
 	}
 	if hc.Resources.CPURealtimePeriod != 0 && hc.Resources.CPURealtimeRuntime != 0 && hc.Resources.CPURealtimeRuntime > hc.Resources.CPURealtimePeriod {
-		return fmt.Errorf("cpu real-time runtime cannot be higher than cpu real-time period")
+		return errdefs.InvalidParameter(fmt.Errorf("cpu real-time runtime cannot be higher than cpu real-time period"))
+	}
+	// We're only producing an error when supported to preserve old behavior.
+	// The OCI runtime may still reject the config though.
+	if si.CPUShares {
```

Comment:
This comment is written a bit poorly; "when supported" should probably be something like "if CPU-shares are supported"; let me touch it up.

Reply:
☝️ done 👍